lib/std/jsbigints.nim

```diff
@@ -0,0 +1,235 @@
+  runnableExamples:
+    doAssert (big"555" xor big"2") == big"553"
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".} =
+  runnableExamples:
+    doAssert (big"999" shl big"2") == big"3996"
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".} =
+  runnableExamples:
+    doAssert (big"999" shr big"2") == big"249"
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".} =
+  runnableExamples:
+    doAssert -(big"10101010101") == big"-10101010101"
+
+func inc*(a: var JsBigInt; b: JsBigInt) {.importjs: "([#][0][0] += #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    inc big1, big2
+    doAssert big1 == big"3"
+
+func dec*(a: var JsBigInt; b: JsBigInt) {.importjs: "([#][0][0] -= #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    dec big1, big2
+    doAssert big1 == big"-1"
+
+func `+=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    big1 += big2
+    doAssert big1 == big"3"
+
+func `-=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    big1 -= big2
+    doAssert big1 == big"-1"
+
+func `*=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"2"
+    let big2: JsBigInt = big"4"
+    big1 *= big2
+    doAssert big1 == big"8"
+
+func inc*(x: var JsBigInt) {.importjs: "(++[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    inc big1
+    doAssert big1 == big"2"
+
+func dec*(x: var JsBigInt) {.importjs: "(--[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = big"3"
+    dec big1
+    doAssert big1 == big"2"
+
+func `%=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] %= #)".} =
+  ## Same as `x = x mod y`.
+  runnableExamples:
+    var big1: JsBigInt = big"10"
+    big1 %= big"2"
+    doAssert big1 == big"0"
+
+func `/=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] /= #)".} =
+  ## Same as `x = x div y`.
+  runnableExamples:
+    var big1: JsBigInt = big"11"
+    big1 /= big"2"
+    doAssert big1 == big"5"
+
+proc addInt*(result: var string; integer: JsBigInt) =
+  ## Converts `integer` to its string representation and appends it to `result`.
+  runnableExamples:
+    var a = "12345678"
+    let b = big"99999999999999999999999999999999999999"
+    a.addInt b
+    doAssert a == "1234567899999999999999999999999999999999999999"
+  result.add integer.toString
+
+proc `+`*(_: JsBigInt): JsBigInt {.error.} # Can not be used by design.
+  ## **Do NOT use.** https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs
+
+proc low*(_: typedesc[JsBigInt]): JsBigInt {.error.} ## **Do NOT use.**
+
+proc high*(_: typedesc[JsBigInt]): JsBigInt {.error.} ## **Do NOT use.**
+
+
+runnableExamples:
```

Comment:
some of this needs to move to `tests/stdlib/tjsbigints.nim`, so that this `runnableExamples:` focuses on the educational part (how do i use bigint, what are the gotchas) instead of the testing part